pkg/serviceaccount/claims_test.go

```diff
@@ -0,0 +1,216 @@
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "g",
+					Name:      "h",
+					UID:       "k",
+				},
+			},
+			// really fast
+			exp: 0,
+			// nil audience
+			aud: nil,
+
+			sc: &jwt.Claims{
+				Subject:   "system:serviceaccount:a:b",
+				IssuedAt:  jwt.NumericDate(1514764800),
+				NotBefore: jwt.NumericDate(1514764800),
+				Expiry:    jwt.NumericDate(1514764800),
+			},
+			pc: &privateClaims{
+				Kubernetes: kubernetes{
+					Namespace: "a",
+					Svcacct: ref{
+						Name: "b",
+						UID:  "c",
+					},
+					Pod: &ref{
+						Name: "e",
+						UID:  "f",
+					},
+				},
+			},
+		},
+		{
+			// pod
+			sa: core.ServiceAccount{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "a",
+					Name:      "b",
+					UID:       "c",
+				},
+			},
+			pod: &core.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "d",
+					Name:      "e",
+					UID:       "f",
+				},
+			},
+			// empty audience
+			aud: []string{},
+			exp: 100,
+
+			sc: &jwt.Claims{
+				Subject:   "system:serviceaccount:a:b",
+				IssuedAt:  jwt.NumericDate(1514764800),
+				NotBefore: jwt.NumericDate(1514764800),
+				Expiry:    jwt.NumericDate(1514764800 + 100),
+			},
+			pc: &privateClaims{
+				Kubernetes: kubernetes{
+					Namespace: "a",
+					Svcacct: ref{
+						Name: "b",
+						UID:  "c",
+					},
+					Pod: &ref{
+						Name: "e",
+						UID:  "f",
+					},
+				},
+			},
+		},
+		{
+			// secret
+			sa: core.ServiceAccount{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "a",
+					Name:      "b",
+					UID:       "c",
+				},
+			},
+			sec: &core.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "g",
+					Name:      "h",
+					UID:       "k",
+				},
+			},
+			exp: 100,
+			// single member audience
+			aud: []string{"1"},
+
+			sc: &jwt.Claims{
+				Subject:   "system:serviceaccount:a:b",
+				Audience:  []string{"1"},
+				IssuedAt:  jwt.NumericDate(1514764800),
+				NotBefore: jwt.NumericDate(1514764800),
+				Expiry:    jwt.NumericDate(1514764800 + 100),
+			},
+			pc: &privateClaims{
+				Kubernetes: kubernetes{
+					Namespace: "a",
+					Svcacct: ref{
+						Name: "b",
+						UID:  "c",
+					},
+					Secret: &ref{
+						Name: "h",
+						UID:  "k",
+					},
+				},
+			},
+		},
+		{
+			// no obj binding
+			sa: core.ServiceAccount{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "a",
+					Name:      "b",
+					UID:       "c",
+				},
+			},
+			exp: 100,
+			// multimember audience
+			aud: []string{"1", "2"},
+
+			sc: &jwt.Claims{
+				Subject:   "system:serviceaccount:a:b",
+				Audience:  []string{"1", "2"},
+				IssuedAt:  jwt.NumericDate(1514764800),
+				NotBefore: jwt.NumericDate(1514764800),
+				Expiry:    jwt.NumericDate(1514764800 + 100),
+			},
+			pc: &privateClaims{
+				Kubernetes: kubernetes{
+					Namespace: "a",
+					Svcacct: ref{
+						Name: "b",
+						UID:  "c",
+					},
+				},
+			},
+		},
+	}
+	for i, c := range cs {
+		t.Run(fmt.Sprintf("case %d", i), func(t *testing.T) {
```

Comment:
AFAIR in many tests we use `map[string]struct` that allows to have readable names for tests.